src/test/system_tests/ceph_s3_tests/test_ceph_s3.js

```diff
@@ -100,29 +99,35 @@ async function test_worker() {
     }
 }
 
-async function run_single_test(test) {
-    let ceph_args = `S3TEST_CONF=${CEPH_TEST.test_dir}${CEPH_TEST.ceph_config}`;
-    if (S3_CEPH_TEST_SIGV4.includes(test)) {
+async function run_single_test(test_name) {
+    let ceph_args = `S3TEST_CONF=${process.cwd()}/${CEPH_TEST.test_dir}${CEPH_TEST.ceph_config}`;
+    if (S3_CEPH_TEST_SIGV4.includes(test_name)) {
         ceph_args += ` S3_USE_SIGV4=true`;
     }
-    let base_cmd = `${ceph_args} ./${CEPH_TEST.test_dir}${CEPH_TEST.s3_test_dir}virtualenv/bin/nosetests`;
-    if (!S3_CEPH_TEST_OUT_OF_SCOPE_REGEXP.test(test)) {
+    let base_cmd = `${ceph_args} tox ${TOX_ARGS}`;
+    if (!S3_CEPH_TEST_OUT_OF_SCOPE_REGEXP.test(test_name)) {
         try {
-            const test_name = test.replace(S3_CEPH_TEST_STEMS_REGEXP, pref => `${pref.slice(0, -1)}:`); //Match against the common test path
             if (test_name.includes('boto')) {
-                base_cmd = `${ceph_args} ./${CEPH_TEST.test_dir}${CEPH_TEST.s3_test_dir}virtualenv/bin/nosetests -v -s -A 'not fails_on_rgw'`;
+                base_cmd = `${ceph_args} tox ${TOX_ARGS} -- -m 'not fails_on_aws'`;
             }
-            const res = await os_utils.exec(`${base_cmd} ${test_name}`, { ignore_rc: false, return_stdout: true });
-            if (res.indexOf('SKIP') >= 0) {
-                console.warn('Test skipped:', test);
-                testing_status.skip.push(test);
+            const res = await os_utils.exec(`${base_cmd} ${process.cwd()}/${CEPH_TEST.test_dir}${CEPH_TEST.s3_test_dir}${test_name}`, { ignore_rc: false, return_stdout: true });
+            if (res.indexOf('skipped') >= 0) {
+                console.warn('Test skipped:', test_name);
+                testing_status.skip.push(test_name);
             } else {
-                console.info('Test Passed:', test);
-                testing_status.pass.push(test);
+                console.info('Test Passed:', test_name);
+                testing_status.pass.push(test_name);
             }
         } catch (err) {
-            console.error('Test Failed:', test);
-            testing_status.fail.push(test);
+            // tox will exit with code 1 on error regardless of pytest exit code. pytest exit code 5 means no tests ran.
+            // can happen when 'not fails_on_aws' flag is on for some tests (there are no boto3 tests for the test)
+            if (err.stdout.indexOf("exited with code 5") >= 0) {
```

Comment:
Is it the same as `err.message.includes("exited with code 5")`?
If so, I prefer that it be more readable (since we don't need the index inside the string).

Reply:
I just used the same function that was used for the skipped tests. I can check if it works and fix it for both of them

Reply:
Did it work?